File: programacion/java/2_unidad/1/Ejercicio2_3.java
import java.util.Scanner;
public class Ejercicio2_3 {
    public static void main(String[] args) {
        Scanner keyboard = new Scanner(System.in);

        System.out.print("Introduce el valor n de la sucesión de Fibonacci: ");
        int pos = Integer.parseInt(keyboard.nextLine());

        if (pos < 0) {
            System.out.println("La posición debe ser 0 o mayor");
        } else if (pos == 0) {
            System.out.println("Posición 0: 0");
        } else if (pos == 1) {
            System.out.println("Posición 1: 1");
        } else {
            int penultimo = 0;
            int ultimo = 1;

            for (int i =2; i <= pos; i++) {
                int aux = penultimo +  ultimo;
                penultimo = ultimo;
                ultimo = aux;
            }
            System.out.println("Posición " + pos + ": " + ultimo);
        }
    }
}

File: programacion/python/unidad_2/Ejercicio2_3.py
pos = int(input("Introduce el valor n de la sucesión de Fibonacci: "))

if pos < 0:
    print("La posición debe ser 0 o mayor.")
elif pos == 0:
    print("Posición 0: 0")
elif pos == 1:
    print("Posición 1: 1")
else:
    penultimo = 0
    ultimo =1
    for i in range(2, pos+1):
        aux = ultimo + penultimo
        penultimo = ultimo
        ultimo = aux
    print(f"Posición {pos}: {ultimo}")
print("Fin de programa.")